src/attr/_make.py

```diff
@@ -343,12 +343,70 @@ def _counter_getter(e):
     return e[1].counter
 
 
-def _transform_attrs(cls, these, auto_attribs, kw_only):
+def _collect_base_attrs(cls, taken_attr_names):
+    """
+    Collect attr.ibs from base classes of *cls*, except *taken_attr_names*.
+    """
+    base_attrs = []
+    base_attr_map = {}  # A dictionary of base attrs to their classes.
+
+    # Traverse the MRO and collect attributes.
+    for base_cls in reversed(cls.__mro__[1:-1]):
+        for a in getattr(base_cls, "__attrs_attrs__", []):
+            if not a.inherited and a.name not in taken_attr_names:
+                a = a._assoc(inherited=True)
+                base_attrs.append(a)
+                base_attr_map[a.name] = base_cls
```

Comment:
It's probably a matter of taste but would you consider reducing the indentation?
```suggestion
            if a.inherited or a.name in taken_attr_names:
                continue
            a = a._assoc(inherited=True)
            base_attrs.append(a)
            base_attr_map[a.name] = base_cls
```

Reply:
It's very much my taste; as you may have guessed there used to be more code and it just stayed that way.